lib/ProductCard/ProductCard.js

```diff
@@ -0,0 +1,96 @@
+import clsx from 'clsx';
+import { ImageGallery } from '../ImageGallery';
+import { StarRating } from '../StarRating';
+
+const ProductCard = (props) => {
+  const {
+    classNames = {},
+    title,
+    price,
+    url,
+    imageUrl,
+    altImage,
+    Image: ImageComponent,
+    starRating = 0,
+    reviews,
+    onAddToCart,
+    displayVariants = false,
+    variantImages = [],
+  } = props;
+
+  return (
+    <div className={clsx('max-w-[282px] min-w-[160px] group', classNames.root)}>
+      <div
+        className={clsx(
+          'bg-slate-100 w-full overflow-hidden',
+          classNames.imageWrapper,
+        )}
+      >
+        {ImageComponent ? (
+          <ImageComponent />
+        ) : (
+          <>
+            {displayVariants && (
+              <div className="hidden w-full group-hover:block">
+                <ImageGallery
+                  classNames={{
+                    thumbnailsReel: 'max-w-[282px] min-w-[160px]',
+                  }}
+                  images={variantImages}
+                  reelPosition="bottom"
+                  navigateOnHover
+                />
+              </div>
+            )}
+            <img
+              className={clsx(
+                'object-cover w-full h-full ',
+                { 'group-hover:hidden': displayVariants },
+                classNames.image,
+              )}
+              src={imageUrl}
+              alt={altImage}
+            />
+          </>
+        )}
+      </div>
+      <div className="py-6 bg-white">
+        <div className="text-center max-w-[225px] min-w-[160px] mx-auto space-y-4 px-3">
+          {/* TODO: Add support for next/link */}
```

Comment:
Let's add support for the custom link element before we merge it. Super important for usage in SPAs.